frontend/src/Router.jsx

```diff
@@ -1,18 +1,42 @@
 import { Route, Routes } from 'react-router-dom'
 import Landing from './pages/landing/LandingController'
+import SignIn from './pages/signIn/SignInController'
 import Post from './pages/post/PostPageController'
 import Follows from './pages/follows/FollowsPageController'
 import Registration from './pages/registration/RegistrationPagesController'
 import User from './pages/user/UserPageController'
+import { AuthContext } from './contexts/AuthProvider'
+import { useContext, useEffect } from 'react'
 
-const Router = () => (
-    <Routes>
-        <Route path="/" element={<Landing />} />
-        <Route path="/post/:id" element={<Post />} />
-        <Route path="/signUp" element={<Registration />} />
-        <Route path="/user/:username/follows" element={<Follows />} />
-        <Route path="/user/:username" element={<User />} />
-    </Routes>
-)
+const Router = () => {
+    // fetch the token to check if the user is authenticated
+
+    let routes
+    const { isAuthenticated } = useContext(AuthContext)
+    const token = localStorage.getItem('token')
+
+    useEffect(() => {}, [isAuthenticated]) // enables rerendering when token is updated
+
+    if (token) {
+        // Auth routes
+        routes = (
+            <Routes>
+                <Route path="/post/:id" element={<Post />} />
+                <Route path="/user/:username/follows" element={<Follows />} />
+            </Routes>
+        )
+    } else {
+        // Unauth routes
+        routes = (
+            <Routes>
+                <Route path="/" element={<Landing />} />
```

Comment:
You should add a Route to "*" here that redirects the user back to "/", so if they try to access something they shouldn't it will just take them to the landing